hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java

```diff
@@ -2209,7 +2237,8 @@ public FSDataOutputStreamBuilder createFile(final Path path) {
       builder
           .create()
           .overwrite(true)
-          .must(FS_S3A_CREATE_PERFORMANCE, performanceCreation);
+          .must(FS_S3A_CREATE_PERFORMANCE,
+              getPerformanceFlags().enabled(PerformanceFlagEnum.Create));
```

Comment:
just a thought not related to this patch, but shall we make `FSDataOutputStreamBuilder` more specific for s3afs 
e.g.
 `public FSDataOutputStreamBuilder<FSDataOutputStream, CreateFileBuilder> createFile(final Path path)`

Reply:
no worries, it was just a minor thing i was thinking about.